Sources/FluentKit/Properties/Parent.swift

```diff
@@ -1,69 +1,132 @@
 @propertyWrapper
-public final class Parent<To>
-    where To: Model
-{
-    @Field
-    public var id: To.IDValue
+public final class Parent<To> where To: ParentRelatable {
+    internal var eagerLoadedValue: EagerLoaded
+    internal var eagerLoadRequest: EagerLoadRequest
+    @Field public var id: To.StoredIDValue
 
     public var wrappedValue: To {
         get {
-            guard let value = self.eagerLoadedValue else {
-                fatalError("Parent relation not eager loaded, use $ prefix to access")
+            switch self.eagerLoadedValue {
+            case .notLoaded: fatalError("Parent relation not eager loaded, use $ prefix to access")
+            case let .loaded(value): return value
             }
-            return value
         }
-        set { fatalError("use $ prefix to access") }
-    }
-
-    public var projectedValue: Parent<To> {
-        return self
+        set {
```

Comment:
Why have a setter at all instead of leaving `wrappedValue` read-only?

Reply:
@tanner0101 You should be the one to answer this.

Reply:
@gwynne the idea is that we wanna eventually support mutating relations in a "swifty" way. If we make this get-only, then you must declare the relations as `let foos: [Foo]` on your model. That makes it so that adding a setter in the future is a breaking change.